discord/user.py

```diff
@@ -51,42 +68,42 @@ class BaseUser(_UserTag):
         bot: bool
         system: bool
 
-    def __init__(self, *, state, data):
-        self._state = state
+    def __init__(self, *, state: ConnectionState, data: UserPayload) -> None:
+        self._state: ConnectionState = state
```

Comment:
I believe `_state` is inferred since `state` is typed, so typing `_state` is not needed